base/record_replay.cc

```diff
@@ -581,4 +582,44 @@ void DoResetPaintSurface() {
   }
 }
 
+std::vector<std::string> *gPseudoStack = nullptr;
+
+static std::vector<std::string>* GetPseudoStack() {
+  if (!gPseudoStack) {
+    gPseudoStack = new std::vector<std::string>();
+  }
+  return gPseudoStack;
+}
+
+size_t PushPseudoStackFrame(const char* name) {
+  std::vector<std::string>* pseudoStack = GetPseudoStack();
+  pseudoStack->push_back(name);
+  return pseudoStack->size();
+}
+void PopPseudoStackFrame(const char* name, size_t id) {
+  CHECK(gPseudoStack != nullptr);
+  CHECK(gPseudoStack->size() == id);
+  CHECK(gPseudoStack->back() == name);
+  gPseudoStack->pop_back();
+}
+std::vector<std::string> ReadPseudoStack() {
```

Comment:
Do we really want this to be a copy?

Reply:
Yeah.. I can change this to return a reference.